packages/rpc-subscriptions-spec/src/__tests__/rpc-subscriptions-pubsub-plan-test.ts

```diff
@@ -304,5 +304,36 @@ describe('executeRpcPubSubSubscriptionPlan', () => {
                 );
             });
         });
+        describe('and then acknowledges a subsequent subscription with a different subscription id', () => {
+            let secondAbortController: AbortController;
+            let priorSubscriptionId: number;
+            beforeEach(async () => {
+                jest.useFakeTimers();
+                priorSubscriptionId = expectedSubscriptionId;
+                secondAbortController = new AbortController();
+                executeRpcPubSubSubscriptionPlan({
+                    channel: mockChannel as RpcSubscriptionsChannel<unknown, unknown>,
+                    signal: secondAbortController.signal,
+                    subscribeRequest: { methodName: 'thingSubscribe', params: [] },
+                    unsubscribeMethodName: 'thingUnsubscribe',
+                }).catch(() => {});
+                await jest.runAllTimersAsync();
+                receiveMessage({ id: lastMessageId, jsonrpc: '2.0', result: (expectedSubscriptionId = 456) });
+            });
+            it('sends the unsubscribe message when the prior subscription aborts', () => {
+                mockSend.mockClear();
+                abortController.abort();
+                expect(mockSend).toHaveBeenCalledWith(
+                    expect.objectContaining({ method: 'thingUnsubscribe', params: [priorSubscriptionId] }),
```

Comment:
This acts as a regression test for #603; tests that the tracker can track two subscriptions independently.